### PR TITLE
Feature/implement activation patching

### PR DESCRIPTION
**Pull request by llinauer**
Add script for demonstration of activation patching in the IOI task

### CODE REVIEW

File: scripts/activation_patching_ioi.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# first, do a forward pass on the corrupted prompt to get the corrupted logits
+patchscope.source.prompt = corrupted_prompt
+patchscope.source_forward_pass()
+corrupted_logits = patchscope.source_output.detach()
+
+
+# then, do a forward pass on the clean prompt to get the clean logits
+patchscope.source.prompt = clean_prompt
+patchscope.source_forward_pass()
+clean_logits = patchscope.source_output.detach()
```
Comment (tvhong) [inline scripts/activation_patching_ioi.py:66]: Do we need to do this hack to get the correct results? Can we use a single `.run()` to get logits from source and target?

Comment (tvhong) [reply scripts/activation_patching_ioi.py:66]: Just noticed that nnsight supports batching with multiple `invoke` contexts

```
with model.trace() as tracer:

  with tracer.invoke('The Eiffel Tower is in the city of'):

    # Ablate the last MLP for only this batch.
    model.transformer.h[-1].mlp.output[0][:] = 0

    # Get the output for only the intervened on batch.
    token_ids_intervention = model.lm_head.output.argmax(dim=-1).save()

  with tracer.invoke('The Eiffel Tower is in the city of'):

    # Get the output for only the original batch.
    token_ids_original = model.lm_head.output.argmax(dim=-1).save()

print("Original token IDs:", token_ids_original)
print("Intervention token IDs:", token_ids_intervention)

print("Original prediction:", model.tokenizer.decode(token_ids_original[0][-1]))
print("Intervention prediction:", model.tokenizer.decode(token_ids_intervention[0][-1]))
```
https://nnsight.net/notebooks/tutorials/walkthrough/

I think patchscope should also support batching. That way, we can get all the results here with a single forward pass.
